tests/std/tests/P0896R4_ranges_alg_unique_copy/test.cpp

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <ranges>
+#include <span>
+#include <utility>
+
+#include <range_algorithm_support.hpp>
+using namespace std;
+
+struct instrumentedPair {
+    pair<int, int> _val         = {0, 0};
+    mutable int _numProjections = 0;

```

Comment:
This is causing the tests to fail, due to `constexpr` incompatibility.

Reply:
I mean it worked in MSVC.

I was so surprised that I wanted to try it

Reply:
Filed a ticket for the [compiler bug](https://developercommunity.visualstudio.com/content/problem/1113613/mutable-subobject-accessible-in-constexpr-context.html)

Reply:
So we learned 2 things from the bug report:

1. @CaseyCarter has a doppelgänger reporting gcc bugs.
2. It turns out clang is (likely) wrong here.

The interesting strandard clause is [this](https://eel.is/c++draft/expr.const#4.8). But as @CaseyCarter points out the term "usable in constant expressions" only applies here if the lifetime of the variable began outside of the constant expression.

Here we have a `static_assert()` so the lifetime of variable `instantiator` starts inside the constant expression.

What could be a bit tricky is the use of e.g. `static constexpr instrumentedPair expectedOutput[4]` because I am unsure when its lifetime began wrt to the constant expression. And we use them to check whether we used the right strategy

Reply:
> What could be a bit tricky is the use of e.g. `static constexpr instrumentedPair expectedOutput[4]` because I am unsure when its lifetime began wrt to the constant expression.

Yes, these were the problem. I have crushed `instrumentedPair` via the sneaky tactic of using the non-key `pair` member to store the projection count so we can use plain `pair<int, int>`.